checkov/terraform/checks/resource/azure/NetworkWatcherFlowLogPeriod.py

```diff
@@ -16,7 +16,9 @@ def scan_resource_conf(self, conf):
             retention_block = conf['retention_policy'][0]
             if retention_block['enabled'][0]:
                 retention_in_days = force_int(retention_block['days'][0])
-                if retention_in_days and retention_in_days >= 90:
+                # Comparing to 0 is necessary since `force_int` may return None (falsey) if
+                # "days" is not a valid integer.
+                if retention_in_days == 0 or retention_in_days >= 90:
```

Comment:
I am not sure that handling `None` in this way would be correct. Omitting `days` would cause the above line that does `['days']` to raise a `KeyError` (perhaps that should be `.get('days', [None])`?). If `days` is present but cannot be converted to an `int`, then `force_int` will have returned `None` and therefore, this expression will evaluate to `False`, the condition will fail, and we'll fall to the `return CheckResult.FAILED` below.

Accepting `None` as a `PASSED` state would be explicitly _incorrect_ because it means that `days` is a totally invalid value and is especially not `0` or a number >= 90.

The only correct way to handle `None` would be a second `if retention_in_days is None: return CheckResult.FAILED` but that's redundant with the final statement in the function.

As it is, omitting `days` does cause an odd `KeyError` failure in the tests; however, it does return what I would expect to be correct error in the CLI execution as a whole, reporting that it failed to run the check.

Basically, the check _should not_ pass without `days` present, because per https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_watcher_flow_log#days, `days` is a required value. There is not a valid "default" value per the documentation. If there were, then perhaps accepting `None` would be correct but to not provide `days` results in an invalid Terraform resource.

Is there a better way to reword the comment that would make this more clear? Should the comment highlight that `None` is not a passing value?

Reply:
Updated the code to _explicitly_ reject `None`. Hopefully this makes the intent clearer. It leaves the `KeyError` as a possibility but that would have happened regardless. Could be made to be `.get('days', [None])` if desired but I don't know if that would be correct since the resource would be invalid.

Otherwise, hopefully explicitly rejecting `None` makes this a little bit clearer with the intent.